pupil-api/src/services/redis-pupil-auth.service.ts

```diff
@@ -37,7 +37,9 @@ export class RedisPupilAuthenticationService implements IPupilAuthenticationServ
     azureQueueService.addMessage('pupil-login', pupilLoginMessage)
     const checkStartedLookupKey = this.buildCheckStartedLookupKey(preparedCheckEntry.checkCode)
     await this.redisService.setex(checkStartedLookupKey, cacheKey, this.eightHoursInSeconds)
-    await this.redisService.expire(cacheKey, config.RedisPreparedCheckExpiryInSeconds)
+    if (preparedCheckEntry.config.practice === false) {
```

Comment:
Regarding Jon's comment for this, is it worth checking if the `config` has a property `practice` using something like `hasOwnProperty` or if it is `undefined`?

Reply:
that's a good point George, because typescript will suggest that if its a non-optional boolean, there is no point checking for undefined.  However in this case, we know it could be undefined at runtime, causing it to go into live mode, just by not being present, which is not what we want.  will add some structure to this. thanks

Reply:
i've added unit tests for the above case.  the existing line is still valid, even when `config.practice` is undefined or non existent